### ISSUE TITLE
Undefined variable and headers already sent errors Paypal Pro (payflow)

### ISSUE THREAD

**Issue by drewvinci** [angelleye/paypal-woocommerce#421]
Started getting these errors with the Pro Payflow class after moving to a dedicated server:

[19-Feb-2016 00:04:18 UTC] PHP Notice:  Undefined variable: card_expiry in /home/rpsiteadmin/public_html/wp-content/plugins/paypal-for-woocommerce/classes/wc-gateway-paypal-pro-payflow-angelleye.php on line 748
[19-Feb-2016 00:04:18 UTC] PHP Notice:  Undefined variable: card_expiry in /home/rpsiteadmin/public_html/wp-content/plugins/paypal-for-woocommerce/classes/wc-gateway-paypal-pro-payflow-angelleye.php on line 771
[19-Feb-2016 00:04:18 UTC] PHP Warning:  session_start(): Cannot send session cookie - headers already sent by (output started at /home/rpsiteadmin/public_html/wp-content/plugins/paypal-for-woocommerce/classes/wc-gateway-paypal-pro-payflow-angelleye.php:748) in /home/rpsiteadmin/public_html/wp-content/plugins/paypal-for-woocommerce/classes/wc-gateway-paypal-pro-payflow-angelleye.php on line 253
[19-Feb-2016 00:04:18 UTC] PHP Warning:  session_start(): Cannot send session cache limiter - headers already sent (output started at /home/rpsiteadmin/public_html/wp-content/plugins/paypal-for-woocommerce/classes/wc-gateway-paypal-pro-payflow-angelleye.php:748) in /home/rpsiteadmin/public_html/wp-content/plugins/paypal-for-woocommerce/classes/wc-gateway-paypal-pro-payflow-angelleye.php on line 253

Here is the system report if needed:

### WordPress Environment ###

Home URL: http://renaissanceperiodization.com
Site URL: http://renaissanceperiodization.com
WC Version: 2.5.2
Log Directory Writable: ✔  /home/rpsiteadmin/public_html/wp-content/uploads/wc-logs/
WP Version: 4.4.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.2.31 (Unix) mod_ssl/2.2.31 OpenSSL/1.0.1e-fips mod_bwlimited/1.4 mod_fcgid/2.3.9
PHP Version: 5.4.45
PHP Post Max Size: 512 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
SUHOSIN Installed: –
MySQL Version: 5.5.48
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 2.5.2
: 
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_termmeta: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔

### Active Plugins (3) ###

Genesis Connect for WooCommerce: by StudioPress – 0.9.8
PayPal for WooCommerce: by Angell EYE – 1.1.9.2
WooCommerce: by WooThemes – 2.5.2

### Settings ###

Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2

### API ###

API Enabled: ✔
API Version: 3.1.0

### WC Pages ###

Shop Base: #833 - /shop/
Cart: #834 - /cart/
Checkout: #835 - /checkout/
My Account: #836 - /my-account/

### Taxonomies ###

Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

### Theme ###

Name: Strength Pro Theme
Version: 1.2
Author URL: http://www.skewld.com
Child Theme: ✔
Parent Theme Name: Genesis
Parent Theme Version: 2.2.6
Parent Theme Author URL: http://www.studiopress.com/
WooCommerce Support: ✔

### Templates ###

Overrides: strength/woocommerce/archive-product.php
strength/woocommerce/single-product.php

**Comment by angelleye** [comment]
It looks like you're having a problem with the session data, which is why the undefined variables are happening.

Can you try disabling the Genesis Connection plugin and see if that helps anything?  Also try switching to a basic theme and see if that helps.

Those "headers already sent" errors you are seeing typically means there is a problem like extra white space somewhere mixed between HTML and PHP code in a page.  My initial thought is that it must be coming from a plugin or theme, so please give try disabling those so we can eliminate that as a possible issue.

**Comment by drewvinci** [comment]
That was my thought as well, but when that didn't work I came here. lol. After seeing your comments, I turned to the only things that were left from the old server, perhaps something in the database? I dug through and found a bunch of old stuff, plugin data long since uninstalled, Optimizepress crap everywhere which was enough for me to test without their data. Rolled up a new cpanel, installed WP, WC + PP4WC, set them up and ran a test purchase like its supposed to be. 

Thanks for your help, sorry to have wasted your time on this. Still don't know exactly whats the cause but I can do selective piece by piece data moves just with the stuff needed, all of the store stuff, users, etc... just do all fresh plugin installs and setups and test in between. I'll either find it or I won't, either way good to go.
